tests/backend-storage/src/test/ServerStorage.test.ts

```diff
@@ -442,6 +442,20 @@ describe(`${ServerStorage.name}: ${serverStorage.constructor.name}`, () => {
     });
   });
 
+  describe(`${serverStorage.listDirectoriesAsync.name}()`, () => {
+    it("should not list more than maxPageSize directories", async () => {
+      await testDirectoryManager.createNew();
+      await testDirectoryManager.createNew();
+      await testDirectoryManager.createNew();
+      const maxPageSize = 2;
+      const queriedDirectories = await serverStorage.listDirectoriesAsync(
+        maxPageSize
+      );
+
+      expect(queriedDirectories[0].length).to.be.lte(maxPageSize);
```

Comment:
When testing paged queries, you should test that you can successfully finish querying everything. Querying just first page doesn't check if you are handling second and last pages correctly.

Reply:
The changes still do not verify that you actually queried more than a single page, only that every single page hasn't been over the limit (there might be more directories than what you created though, so it should just be greater or equal comparison).

Also the same test should be added for objects.

Reply:
Yes actually due to remnants artefacts from other integration tests, I can't really test the way I would prefer to. But actually since the page iterator code is used in the listObjects, listDirectories, it already checks that everything was queried.